### PR TITLE
Convert docs backend from Sphinx (reStructuredText) to MkDocs (Markdown)

### PR DESCRIPTION
**Pull request by philiplinden**
This PR overhauls the documentation to use [MkDocs](https://www.mkdocs.org/) rather than [Sphinx](https://www.sphinx-doc.org/en/master/). 

The end result should be 1:1 parity with the existing documentation under Sphinx, but instead using the MkDocs backend with documentation pages written in markdown.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -192,4 +192,7 @@ TSWLatexianTemp*
 .DS_Store
 
 # Sphinx
-docs/_build
\ No newline at end of file
+docs/_build
+
+# virtualenv
+venv
```
Comment (dmitchellspace) [inline .gitignore:198]: `site/` should be added here as well

File: docs/contribute.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Edit the docs as Markdown files, then use `mkdocs` to preview them in your
+browser.
+
+```bash
+pip install mkdocs
+```
+
+* `mkdocs serve` - Start the live-reloading docs server.
+* `mkdocs build` - Build the documentation site.
+* `mkdocs -h` - Print help message and exit.
```
Comment (dmitchellspace) [inline docs/contribute.md]: Would help to also link https://www.mkdocs.org/ here. Super mega bonus points if you include a gif here of the actual workflow.

File: .github/workflows/publish-docs.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    branches:
+      - master
+
+jobs:
+  build:
+    name: Deploy Docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout master
+        uses: actions/checkout@v1
```
Comment (dmitchellspace) [inline .github/workflows/publish-docs.yml]: It looks like v2 is the latest for this. Any particular reason we are using v1?

See: 
https://github.com/actions/checkout
https://github.com/actions/checkout/blob/master/CHANGELOG.md

File: .github/workflows/publish-docs.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+      - name: Deploy docs
+        uses: mhausenblas/mkdocs-deploy-gh-pages@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          CUSTOM_DOMAIN: terrahab.github.io
```
Comment (dmitchellspace) [inline .github/workflows/publish-docs.yml:19]: Has this workflow been tested successfully?

**Review by dmitchellspace — CHANGES_REQUESTED** [review]
See comments below. I'm mainly concerned about the success of the workflow and `site/` being added to .gitignore

File: .github/workflows/publish-docs.yml (hunk 4 above)
Comment (philiplinden) [reply .github/workflows/publish-docs.yml:19]: I installed https://github.com/nektos/act to test it but I'm not sure if it's working. Gonna test some more

Comment (dmitchellspace) [reply .github/workflows/publish-docs.yml:19]: Excellent! Once you're good with this, go ahead and rebase to squash these to a single commit and we can merge

Comment (philiplinden) [reply .github/workflows/publish-docs.yml:19]:
```
✦ ❯ act -l
 ╭─────────────╮
 │ Deploy Docs │
 ╰─────────────╯

TerraMICRO on  mkdocs 
✦ ❯ act
[Publish Docs via GitHub Pages/Deploy Docs] 🚀  Start image=node:12.6-buster-slim
[Publish Docs via GitHub Pages/Deploy Docs]   🐳  docker run image=node:12.6-buster-slim entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Publish Docs via GitHub Pages/Deploy Docs]   🐳  docker cp src=/home/phil/repos/terrahab/TerraMICRO/. dst=/github/workspace
[Publish Docs via GitHub Pages/Deploy Docs] ⭐  Run Checkout master
[Publish Docs via GitHub Pages/Deploy Docs]   ✅  Success - Checkout master
[Publish Docs via GitHub Pages/Deploy Docs] ⭐  Run Deploy docs
[Publish Docs via GitHub Pages/Deploy Docs]   ☁  git clone 'https://github.com/mhausenblas/mkdocs-deploy-gh-pages' # ref=master
[Publish Docs via GitHub Pages/Deploy Docs]   🐳  docker build -t act-mhausenblas-mkdocs-deploy-gh-pages-master:latest /home/phil/.cache/act/mhausenblas-mkdocs-deploy-gh-pages@master
[Publish Docs via GitHub Pages/Deploy Docs]   🐳  docker run image=act-mhausenblas-mkdocs-deploy-gh-pages-master:latest entrypoint=[] cmd=[]
| INFO: Setting custom domain for github pages
| nektos/act
| nektos/act@users.noreply.github.com
| error: refs/heads/master does not point to a valid object!
| error: refs/remotes/origin/HEAD does not point to a valid object!
| error: refs/remotes/origin/master does not point to a valid object!
| INFO    -  Cleaning site directory 
| INFO    -  Building documentation to directory: /github/workspace/site 
| INFO    -  Documentation built in 0.15 seconds 
| INFO    -  Copying '/github/workspace/site' to 'gh-pages' branch and pushing to GitHub. 
| ERROR   -  Failed to deploy to GitHub with error: 
| error: refs/heads/master does not point to a valid object!
| error: refs/remotes/origin/HEAD does not point to a valid object!
| error: refs/remotes/origin/master does not point to a valid object!
| error: refs/heads/master does not point to a valid object!
| error: refs/remotes/origin/HEAD does not point to a valid object!
| error: refs/remotes/origin/master does not point to a valid object!
| fatal: no path specified; see 'git help pull' for valid url syntax
|  
[Publish Docs via GitHub Pages/Deploy Docs]   ❌  Failure - Deploy docs
Error: exit with `FAILURE`: 1
```

I say we roll with it. 

Squash and merge. This branch was created from the latest commit from master so you shouldn't need to rebase.

Comment (dmitchellspace) [reply .github/workflows/publish-docs.yml:19]: Can you squash without rebasing? I usually follow this flow: https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec

Comment (dmitchellspace) [reply .github/workflows/publish-docs.yml:19]: Either way, it's done now. I'll go ahead and merge it

**Pull request merged**